### PR TITLE
i2c: Bug fixes for default RTIO handler

### PR DESCRIPTION
**Pull request by teburd**
TXRX is meant specifically to handle a full duplex bus like SPI, I2C is half duplex meaning only read or write can be performed at once. Drop TXRX from the default handler.

Each transaction should result in a single call to i2c_transfer. This corrects the default handler to do just that. Also simplifies the various helpers here.

### CODE REVIEW

**Review by ubieda — DISMISSED** [review]

**Comment by ubieda** [comment]
Not sure why this PR didn't get review-requests.

**Comment by XenuIsWatching** [comment]
looks like some tests may need to be updated... but I ported over this code to my default i3c rtio handler branch that I'm working on and it's looking as expected with 'restarts' in between transactions. Thanks for this!!

**Review by ubieda — DISMISSED** [review]
Code looks good! Approving assuming we take care of https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/subsys/rtio/rtio_i2c.

File: drivers/i2c/i2c_rtio_default.c [hunk 1]
```diff
@@ -12,97 +12,115 @@
...
+		txn_last = rtio_txn_next(txn_last);
+	} while (rc == 0 && txn_last != NULL);
+
+	if (rc != 0) {
+		rtio_iodev_sqe_err(txn_first, rc);
+		return;
+	}
+
+	/* Allocate msgs on the stack */
+	struct i2c_msg msgs[num_msgs];
```
Comment (bjarki-andreasen) [inline drivers/i2c/i2c_rtio_default.c:89]: Variable length array types are not MISRA compliant :) not sure why CI is not complaining :) I ran into issues with it here https://github.com/zephyrproject-rtos/zephyr/pull/77552

Comment (teburd) [reply drivers/i2c/i2c_rtio_default.c:89]: I'm open to suggestions here... I didn't want to bother with heap. I guess a statically sized array on the stack? painful.

Comment (ubieda) [reply drivers/i2c/i2c_rtio_default.c:89]: How about a Memslab Pool: You'd only take it while performing the task, so hopefully is low-ish risk?

File: drivers/i2c/i2c_rtio_default.c [hunk 2]
```diff
@@ -12,97 +12,126 @@
...
+		rtio_iodev_sqe_err(txn_first, rc);
+		return;
+	}
+
+	/* Allocate msgs on the stack, MISRA doesn't like VLAs so we need a statically
+	* sized array here. It's pretty unlikely we have more than 4 i2c messages
+	* in a transaction as we typically would only have 2, one to write a
+	* register address, and another to read/write the register into an array
+	*/
+	if (num_msgs > 4) {
```
Comment (bjarki-andreasen) [inline drivers/i2c/i2c_rtio_default.c:94]: maybe overlay pedantic, but maybe define the 4? `#define RTIO_I2C_MSG_ALLOC 4` and even better, make it a Kconfig?

File: drivers/i2c/i2c_rtio_default.c (hunk 1 above)
Comment (teburd) [reply drivers/i2c/i2c_rtio_default.c]: Made it a Kconfig as suggested by @bjarki-andreasen with a default of 4

File: drivers/i2c/i2c_rtio_default.c (hunk 2 above)
Comment (teburd) [reply drivers/i2c/i2c_rtio_default.c]: yes, done

**Review by ubieda — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]
Nice!

**Pull request merged**